### ISSUE TITLE
Add library search paths

### ISSUE THREAD

**Issue by jmoguillansky-gpsw** [mesonbuild/meson#8091]
**Describe the bug**
Meson cc.find_library doesn't support adding library search paths, only overriding system library search path.

**To Reproduce**
call cc.find_library with a system library, and modify the dirs variable.

**Expected behavior**
Please provide the ability to 
1) query the system library search paths
and/or
2) add library search paths to existing search paths

**system parameters**

**Comment by amitdo** [comment]
Maybe using the environment variable `LIBRARY_PATH` can help here.

**Comment by jmoguillansky-gpsw** [comment]
We should be able to add library search paths, not override

**Comment by amitdo** [comment]
`export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib`

**Comment by jmoguillansky-gpsw** [comment]
I'm talking about windows build

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by ptyx092** [comment]
maybe you can try c_args option:
meson setup build -Dc_args=-I/opt/local/include

**Comment by ptyx092** [comment]
maybe you can try c_args option:
meson setup build -Dc_args=-L/opt/local/lib

**Comment by ngie-eign** [comment]
Yeah, it looks complicated because meson doesn’t have a simple conversion routine for supporting standard environment variables like LDFLAGS. I think this is great progress though, so I think this should be considered for merge as-is and maybe I can come up with an alternate interface using pkg-config, etc, that makes sense.
Ref: https://stackoverflow.com/questions/61058722/how-can-i-specify-library-path-when-using-meson

**Comment by ngie-eign** [comment]
@eli-schwartz : the problem with LIBRARY_PATH, based on my quick reading of the above Stack Overflow post is that it only handles resolution of libraries. -Wl,-rpath needs to be supplied to the linker so the library can be found post-compilation, unless the second recipe in the accepted solution is employed.

**Comment by Rodrigodd** [comment]
Was hitting this for `depency(iconv)`. Got it to work by passing `-Dc_args=-I/path/to/include` and `-Dc_link_args=-L/path/to/lib`. In my case I am linking `libiconv.a` statically, so I don't had problem with `-Wl,-rpath`.

I have tried setting `LIBRARY_PATH`, but didn't work for me, maybe because I am using `clang`.
